linkerd/app/src/lib.rs

```diff
@@ -188,24 +188,36 @@ impl Config {
             })
         }?;
 
-        debug!(config = ?oc_collector, "Building client");
-        let oc_collector = {
+        debug!(config = ?trace_collector, "Building client");
+        let trace_collector = {
             let control_metrics =
-                ControlMetrics::register(registry.sub_registry_with_prefix("opencensus"));
+                ControlMetrics::register(if let Some(prefix) = trace_collector.metrics_prefix() {
+                    registry.sub_registry_with_prefix(prefix)
+                } else {
+                    &mut registry
+                });
```

Comment:
I was thinking something along the lines of

```suggestion
                if let Some(prefix) = trace_collector.metrics_prefix() {
                    ControlMetrics::register(registry.sub_registry_with_prefix(prefix))
                } else {
                    ControlMetrics::register(&mut Registry::default())
                };
```

Reply:
Gotcha, wasn't 100% sure what the correct semantics were here.